lib/cloudshaper/module.rb

```diff
@@ -26,12 +26,9 @@ def build_submodule(parent_module, module_name)
     end
 
     def generate_child_module(parent_module)
-      variables = @fields.clone
-      variables.delete(:source)
-      variables[:terraform_stack_id] = parent_module.id
       child_name = @fields[:source].gsub(/^module_/, '')
       child_module = StackModules.get(child_name)
-      child_module.build(variables)
+      child_module.build(cloudshaper_stack_id: parent_module.id)
```

Comment:
A lengthy description below, but the tl;dr is that you pass variables to submodules via arguments in the `submodule` block in the parent module.